### ISSUE TITLE
Wrong Ninja Path (IEP-156)

### ISSUE THREAD

**Issue by axbusch** [espressif/idf-eclipse-plugin#70]
ok, i updated win 7 to win10

i reinstalled eclipse and the idf tools. here is the output of list tools
```
Executing C:\python3_7\python.exe D:\esp\idf\tools\idf_tools.py list
* xtensa-esp32-elf: Toolchain for Xtensa (ESP32) based on GCC
  - esp-2019r2-8.2.0 (recommended, installed)
* esp32ulp-elf: Toolchain for ESP32 ULP coprocessor
  - 2.28.51.20170517 (recommended, installed)
* cmake: CMake build system
  - 3.13.4 (recommended, installed)
* openocd-esp32: OpenOCD for ESP32
  - v0.10.0-esp32-20190313 (recommended, installed)
* mconf: menuconfig tool
  - v4.6.0.0-idf-20190628 (recommended, installed)
* ninja: Ninja build system
  - 1.9.0 (recommended, installed)
* idf-exe: IDF wrapper tool for Windows
  - 1.0.1 (recommended, installed)
* ccache: Ccache (compiler cache)
  - 3.7 (recommended, installed)
```
The folder when i install the tools via Eclipse is
C:\Users\funky\.espressif\tools\

When i want to build my project

>  Building in: D:\esp\remote\esp32_mp3_decoder\build
> cmake --build . -- -v
> Das System kann die angegebene Datei nicht finden
> CMake Error: Generator: execution of make failed. Make command was: "D:/esp/toolchain/.espressif/tools/ninja/1.9.0/ninja.exe" "-v"

"D:/esp/toolchain/.espressif/tools/ninja/1.9.0/" was the old path on the win7 installation but i don't know where this path is still set. where is this path taken from?

**Comment by kolipakakondal** [comment]
Hi @axbusch 
Could you check the PATH from the environment variables?  This might be still having old and new paths. 
`Preferences > C/C++ Build > Environment > PATH`

**Comment by axbusch** [comment]
there was the old path. i removed all the variables, reinstalled the tools but still the same behavior

**Comment by kolipakakondal** [comment]
@axbusch After removing the PATH manually and reinstalling the tools from eclipse - does the PATH still have the old entries?  Could you please provide the complete PATH value?

**Comment by leenowell** [comment]
Hi,

I had exactly this problem and manually fixing it or indeed removing it and reinstalling the tools did not help. I ended up deleting the build folder and rebuilding and it worked fine.  Before deleting the build folder I did a grep for all files in my project with ninja in to try and identify the root cause so hopefully this will help find the bug?

```
4$ grep -r ninja .

binary file ./build/.ninja_deps matches
./build/.ninja_log:# ninja log v5
./build/.ninja_log:74	25774	1611861443674457600	build.ninja	2e2d2707f897cc00
Binary file ./build/bootloader/.ninja_deps matches
./build/bootloader/.ninja_log:# ninja log v5
./build/bootloader/.ninja_log:12	4942	1610485046308597200	build.ninja	70643785a1969473
./build/bootloader/.ninja_log:75	13919	1611435647524265700	build.ninja	70643785a1969473
./build/bootloader/.ninja_log:7	5681	1611861506802916600	build.ninja	70643785a1969473
./build/bootloader/build.ninja:ninja_required_version = 1.5
./build/bootloader/build.ninja:include rules.ninja
./build/bootloader/build.ninja:build build.ninja: RERUN_CMAKE | /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeASMInformation.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeCInformation.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeCXXInformation.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeCommonLanguageInclude.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeGenericSystem.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeInitializeConfigs.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeLanguageInformation.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeSystemSpecificInformation.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeSystemSpecificInitialize.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Compiler/CMakeCommonCompilerMacros.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Compiler/GNU-ASM.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Compiler/GNU-C.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Compiler/GNU-CXX.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Compiler/GNU.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/ExternalProject.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/FindGit.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/FindPackageMessage.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Platform/Generic.cmake /home/lee/esp/esp-idf/.git/HEAD /home/lee/esp/esp-idf/.git/modules/components/asio/asio/HEAD /home/lee/esp/esp-idf/.git/modules/components/bootloader/subproject/components/micro-ecc/micro-ecc/HEAD /home/lee/esp/esp-idf/.git/modules/components/bt/controller/lib/HEAD /home/lee/esp/esp-idf/.git/modules/components/bt/host/nimble/nimble/HEAD /home/lee/esp/esp-idf/.git/modules/components/cbor/tinycbor/HEAD /home/lee/esp/esp-idf/.git/modules/components/coap/libcoap/HEAD /home/lee/esp/esp-idf/.git/modules/components/esp_wifi/lib/HEAD /home/lee/esp/esp-idf/.git/modules/components/esptool_py/esptool/HEAD /home/lee/esp/esp-idf/.git/modules/components/expat/expat/HEAD /home/lee/esp/esp-idf/.git/modules/components/json/cJSON/HEAD /home/lee/esp/esp-idf/.git/modules/components/libsodium/libsodium/HEAD /home/lee/esp/esp-idf/.git/modules/components/lwip/lwip/HEAD /home/lee/esp/esp-idf/.git/modules/components/mbedtls/mbedtls/HEAD /home/lee/esp/esp-idf/.git/modules/components/mqtt/esp-mqtt/HEAD /home/lee/esp/esp-idf/.git/modules/components/nghttp/nghttp2/HEAD /home/lee/esp/esp-idf/.git/modules/components/protobuf-c/protobuf-c/HEAD /home/lee/esp/esp-idf/.git/modules/components/spiffs/spiffs/HEAD /home/lee/esp/esp-idf/.git/modules/components/tinyusb/tinyusb/HEAD /home/lee/esp/esp-idf/.git/modules/components/unity/unity/HEAD /home/lee/esp/esp-idf/.git/modules/examples/build_system/cmake/import_lib/main/lib/tinyxml2/HEAD /home/lee/esp/esp-idf/.git/modules/examples/peripherals/secure_element/atecc608_ecdsa/components/esp-cryptoauthlib/HEAD /home/lee/esp/esp-idf/CMakeLists.txt /home/lee/esp/esp-idf/components/asio/asio/.git /home/lee/esp/esp-idf/components/bootloader/CMakeLists.txt /home/lee/esp/esp-idf/components/bootloader/project_include.cmake /home/lee/esp/esp-idf/components/bootloader/subproject/CMakeLists.txt /home/lee/esp/esp-idf/components/bootloader/subproject/components/micro-ecc/CMakeLists.txt /home/lee/esp/esp-idf/components/bootloader/subproject/components/micro-ecc/micro-ecc/.git /home/lee/esp/esp-idf/components/bootloader/subproject/main/CMakeLists.txt /home/lee/esp/esp-idf/components/bootloader_support/CMakeLists.txt /home/lee/esp/esp-idf/components/bt/controller/lib/.git /home/lee/esp/esp-idf/components/bt/host/nimble/nimble/.git /home/lee/esp/esp-idf/components/cbor/tinycbor/.git /home/lee/esp/esp-idf/components/coap/libcoap/.git /home/lee/esp/esp-idf/components/efuse/CMakeLists.txt /home/lee/esp/esp-idf/components/efuse/esp32/sources.cmake /home/lee/esp/esp-idf/components/esp32/CMakeLists.txt /home/lee/esp/esp-idf/components/esp32/project_include.cmake /home/lee/esp/esp-idf/components/esp_common/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_rom/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_wifi/lib/.git /home/lee/esp/esp-idf/components/esptool_py/CMakeLists.txt /home/lee/esp/esp-idf/components/esptool_py/esptool/.git /home/lee/esp/esp-idf/components/esptool_py/project_include.cmake /home/lee/esp/esp-idf/components/expat/expat/.git /home/lee/esp/esp-idf/components/json/cJSON/.git /home/lee/esp/esp-idf/components/libsodium/libsodium/.git /home/lee/esp/esp-idf/components/log/CMakeLists.txt /home/lee/esp/esp-idf/components/lwip/lwip/.git /home/lee/esp/esp-idf/components/mbedtls/mbedtls/.git /home/lee/esp/esp-idf/components/mqtt/esp-mqtt/.git /home/lee/esp/esp-idf/components/nghttp/nghttp2/.git /home/lee/esp/esp-idf/components/partition_table/CMakeLists.txt /home/lee/esp/esp-idf/components/partition_table/project_include.cmake /home/lee/esp/esp-idf/components/protobuf-c/protobuf-c/.git /home/lee/esp/esp-idf/components/soc/CMakeLists.txt /home/lee/esp/esp-idf/components/soc/soc/CMakeLists.txt /home/lee/esp/esp-idf/components/soc/soc/esp32/CMakeLists.txt /home/lee/esp/esp-idf/components/soc/src/esp32/CMakeLists.txt /home/lee/esp/esp-idf/components/spi_flash/CMakeLists.txt /home/lee/esp/esp-idf/components/spiffs/spiffs/.git /home/lee/esp/esp-idf/components/tinyusb/tinyusb/.git /home/lee/esp/esp-idf/components/unity/unity/.git /home/lee/esp/esp-idf/components/xtensa/CMakeLists.txt /home/lee/esp/esp-idf/examples/build_system/cmake/import_lib/main/lib/tinyxml2/.git /home/lee/esp/esp-idf/examples/peripherals/secure_element/atecc608_ecdsa/components/esp-cryptoauthlib/.git /home/lee/esp/esp-idf/tools/cmake/build.cmake /home/lee/esp/esp-idf/tools/cmake/component.cmake /home/lee/esp/esp-idf/tools/cmake/crosstool_version_check.cmake /home/lee/esp/esp-idf/tools/cmake/dfu.cmake /home/lee/esp/esp-idf/tools/cmake/git_submodules.cmake /home/lee/esp/esp-idf/tools/cmake/idf.cmake /home/lee/esp/esp-idf/tools/cmake/kconfig.cmake /home/lee/esp/esp-idf/tools/cmake/ldgen.cmake /home/lee/esp/esp-idf/tools/cmake/project.cmake /home/lee/esp/esp-idf/tools/cmake/project_description.json.in /home/lee/esp/esp-idf/tools/cmake/targets.cmake /home/lee/esp/esp-idf/tools/cmake/third_party/GetGitRevisionDescription.cmake /home/lee/esp/esp-idf/tools/cmake/third_party/GetGitRevisionDescription.cmake.in /home/lee/esp/esp-idf/tools/cmake/toolchain-esp32.cmake /home/lee/esp/esp-idf/tools/cmake/utilities.cmake /home/lee/esp/esp-idf/tools/cmake/version.cmake /home/lee/esp/esp-idf/tools/kconfig_new/confgen.py /home/lee/esp/esp-idf/tools/kconfig_new/config.env.in /home/lee/esp/esp-idf/tools/kconfig_new/kconfiglib.py /media/Documents/Lee/Dev/SMSworkspace/SmartAlarm4/sdkconfig CMakeCache.txt CMakeFiles/3.13.4/CMakeASMCompiler.cmake CMakeFiles/3.13.4/CMakeCCompiler.cmake CMakeFiles/3.13.4/CMakeCXXCompiler.cmake CMakeFiles/3.13.4/CMakeSystem.cmake CMakeFiles/git-data/grabRef.cmake config/sdkconfig.cmake config/sdkconfig.h
./build/bootloader/CMakeCache.txt://Program used to build from build.ninja files.
./build/bootloader/CMakeCache.txt:CMAKE_MAKE_PROGRAM:FILEPATH=/home/lee/.espressif/tools/ninja/1.9.0/ninja
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_733fa"
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_a5f09"
./build/bootloader/CMakeFiles/CMakeOutput.log:  ignore line: [Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_a5f09"]
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_44c36"
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_84bce"
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_b9ce2"
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_3714c"
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_6bc8c"
./build/bootloader/CMakeFiles/CMakeOutput.log:  ignore line: [Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_6bc8c"]
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_5c5e8"
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_6817f"
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_449d6"
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_0e294"
./build/bootloader/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_f1865"
./build/bootloader/rules.ninja:# It is included in the main 'build.ninja'.
./build/bootloader/rules.ninja:  command = /home/lee/.espressif/tools/ninja/1.9.0/ninja -t clean
./build/bootloader/rules.ninja:  command = /home/lee/.espressif/tools/ninja/1.9.0/ninja -t targets
./build/build.ninja:ninja_required_version = 1.5
./build/build.ninja:include rules.ninja
./build/build.ninja:build build.ninja: RERUN_CMAKE | ../CMakeLists.txt ../main/CMakeLists.txt ../sdkconfig /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeASMInformation.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeCInformation.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeCXXInformation.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeCommonLanguageInclude.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeGenericSystem.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeInitializeConfigs.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeLanguageInformation.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeSystemSpecificInformation.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/CMakeSystemSpecificInitialize.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Compiler/CMakeCommonCompilerMacros.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Compiler/GNU-ASM.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Compiler/GNU-C.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Compiler/GNU-CXX.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Compiler/GNU.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/ExternalProject.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/FindCygwin.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/FindGit.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/FindPackageMessage.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/FindPerl.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/FindPythonInterp.cmake /home/lee/.espressif/tools/cmake/3.13.4/share/cmake-3.13/Modules/Platform/Generic.cmake /home/lee/esp/esp-idf/.git/HEAD /home/lee/esp/esp-idf/.git/modules/components/asio/asio/HEAD /home/lee/esp/esp-idf/.git/modules/components/bootloader/subproject/components/micro-ecc/micro-ecc/HEAD /home/lee/esp/esp-idf/.git/modules/components/bt/controller/lib/HEAD /home/lee/esp/esp-idf/.git/modules/components/bt/host/nimble/nimble/HEAD /home/lee/esp/esp-idf/.git/modules/components/cbor/tinycbor/HEAD /home/lee/esp/esp-idf/.git/modules/components/coap/libcoap/HEAD /home/lee/esp/esp-idf/.git/modules/components/esp_wifi/lib/HEAD /home/lee/esp/esp-idf/.git/modules/components/esptool_py/esptool/HEAD /home/lee/esp/esp-idf/.git/modules/components/expat/expat/HEAD /home/lee/esp/esp-idf/.git/modules/components/json/cJSON/HEAD /home/lee/esp/esp-idf/.git/modules/components/libsodium/libsodium/HEAD /home/lee/esp/esp-idf/.git/modules/components/lwip/lwip/HEAD /home/lee/esp/esp-idf/.git/modules/components/mbedtls/mbedtls/HEAD /home/lee/esp/esp-idf/.git/modules/components/mqtt/esp-mqtt/HEAD /home/lee/esp/esp-idf/.git/modules/components/nghttp/nghttp2/HEAD /home/lee/esp/esp-idf/.git/modules/components/protobuf-c/protobuf-c/HEAD /home/lee/esp/esp-idf/.git/modules/components/spiffs/spiffs/HEAD /home/lee/esp/esp-idf/.git/modules/components/tinyusb/tinyusb/HEAD /home/lee/esp/esp-idf/.git/modules/components/unity/unity/HEAD /home/lee/esp/esp-idf/.git/modules/examples/build_system/cmake/import_lib/main/lib/tinyxml2/HEAD /home/lee/esp/esp-idf/.git/modules/examples/peripherals/secure_element/atecc608_ecdsa/components/esp-cryptoauthlib/HEAD /home/lee/esp/esp-idf/CMakeLists.txt /home/lee/esp/esp-idf/components/app_trace/CMakeLists.txt /home/lee/esp/esp-idf/components/app_trace/project_include.cmake /home/lee/esp/esp-idf/components/app_update/CMakeLists.txt /home/lee/esp/esp-idf/components/asio/CMakeLists.txt /home/lee/esp/esp-idf/components/asio/asio/.git /home/lee/esp/esp-idf/components/bootloader/CMakeLists.txt /home/lee/esp/esp-idf/components/bootloader/project_include.cmake /home/lee/esp/esp-idf/components/bootloader/subproject/components/micro-ecc/micro-ecc/.git /home/lee/esp/esp-idf/components/bootloader_support/CMakeLists.txt /home/lee/esp/esp-idf/components/bt/CMakeLists.txt /home/lee/esp/esp-idf/components/bt/controller/lib/.git /home/lee/esp/esp-idf/components/bt/host/nimble/nimble/.git /home/lee/esp/esp-idf/components/cbor/CMakeLists.txt /home/lee/esp/esp-idf/components/cbor/tinycbor/.git /home/lee/esp/esp-idf/components/coap/CMakeLists.txt /home/lee/esp/esp-idf/components/coap/libcoap/.git /home/lee/esp/esp-idf/components/console/CMakeLists.txt /home/lee/esp/esp-idf/components/cxx/CMakeLists.txt /home/lee/esp/esp-idf/components/driver/CMakeLists.txt /home/lee/esp/esp-idf/components/efuse/CMakeLists.txt /home/lee/esp/esp-idf/components/efuse/esp32/sources.cmake /home/lee/esp/esp-idf/components/esp-tls/CMakeLists.txt /home/lee/esp/esp-idf/components/esp32/CMakeLists.txt /home/lee/esp/esp-idf/components/esp32/project_include.cmake /home/lee/esp/esp-idf/components/esp_adc_cal/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_common/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_eth/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_event/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_gdbstub/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_hid/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_http_client/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_http_server/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_https_ota/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_https_server/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_ipc/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_local_ctrl/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_netif/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_ringbuf/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_rom/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_serial_slave_link/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_system/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_system/port/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_system/port/esp32/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_timer/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_websocket_client/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_wifi/CMakeLists.txt /home/lee/esp/esp-idf/components/esp_wifi/lib/.git /home/lee/esp/esp-idf/components/espcoredump/CMakeLists.txt /home/lee/esp/esp-idf/components/esptool_py/CMakeLists.txt /home/lee/esp/esp-idf/components/esptool_py/esptool/.git /home/lee/esp/esp-idf/components/esptool_py/project_include.cmake /home/lee/esp/esp-idf/components/expat/CMakeLists.txt /home/lee/esp/esp-idf/components/expat/expat/.git /home/lee/esp/esp-idf/components/fatfs/CMakeLists.txt /home/lee/esp/esp-idf/components/freemodbus/CMakeLists.txt /home/lee/esp/esp-idf/components/freertos/CMakeLists.txt /home/lee/esp/esp-idf/components/heap/CMakeLists.txt /home/lee/esp/esp-idf/components/idf_test/CMakeLists.txt /home/lee/esp/esp-idf/components/jsmn/CMakeLists.txt /home/lee/esp/esp-idf/components/json/CMakeLists.txt /home/lee/esp/esp-idf/components/json/cJSON/.git /home/lee/esp/esp-idf/components/libsodium/CMakeLists.txt /home/lee/esp/esp-idf/components/libsodium/libsodium/.git /home/lee/esp/esp-idf/components/log/CMakeLists.txt /home/lee/esp/esp-idf/components/lwip/CMakeLists.txt /home/lee/esp/esp-idf/components/lwip/lwip/.git /home/lee/esp/esp-idf/components/mbedtls/CMakeLists.txt /home/lee/esp/esp-idf/components/mbedtls/mbedtls/.git /home/lee/esp/esp-idf/components/mbedtls/mbedtls/CMakeLists.txt /home/lee/esp/esp-idf/components/mbedtls/mbedtls/include/CMakeLists.txt /home/lee/esp/esp-idf/components/mbedtls/mbedtls/library/CMakeLists.txt /home/lee/esp/esp-idf/components/mdns/CMakeLists.txt /home/lee/esp/esp-idf/components/mqtt/CMakeLists.txt /home/lee/esp/esp-idf/components/mqtt/esp-mqtt/.git /home/lee/esp/esp-idf/components/newlib/CMakeLists.txt /home/lee/esp/esp-idf/components/nghttp/CMakeLists.txt /home/lee/esp/esp-idf/components/nghttp/nghttp2/.git /home/lee/esp/esp-idf/components/nvs_flash/CMakeLists.txt /home/lee/esp/esp-idf/components/openssl/CMakeLists.txt /home/lee/esp/esp-idf/components/partition_table/CMakeLists.txt /home/lee/esp/esp-idf/components/partition_table/partitions_singleapp.csv /home/lee/esp/esp-idf/components/partition_table/project_include.cmake /home/lee/esp/esp-idf/components/perfmon/CMakeLists.txt /home/lee/esp/esp-idf/components/protobuf-c/CMakeLists.txt /home/lee/esp/esp-idf/components/protobuf-c/protobuf-c/.git /home/lee/esp/esp-idf/components/protocomm/CMakeLists.txt /home/lee/esp/esp-idf/components/pthread/CMakeLists.txt /home/lee/esp/esp-idf/components/sdmmc/CMakeLists.txt /home/lee/esp/esp-idf/components/soc/CMakeLists.txt /home/lee/esp/esp-idf/components/soc/soc/CMakeLists.txt /home/lee/esp/esp-idf/components/soc/soc/esp32/CMakeLists.txt /home/lee/esp/esp-idf/components/soc/src/esp32/CMakeLists.txt /home/lee/esp/esp-idf/components/spi_flash/CMakeLists.txt /home/lee/esp/esp-idf/components/spiffs/CMakeLists.txt /home/lee/esp/esp-idf/components/spiffs/project_include.cmake /home/lee/esp/esp-idf/components/spiffs/spiffs/.git /home/lee/esp/esp-idf/components/tcp_transport/CMakeLists.txt /home/lee/esp/esp-idf/components/tcpip_adapter/CMakeLists.txt /home/lee/esp/esp-idf/components/tinyusb/CMakeLists.txt /home/lee/esp/esp-idf/components/tinyusb/tinyusb/.git /home/lee/esp/esp-idf/components/ulp/CMakeLists.txt /home/lee/esp/esp-idf/components/ulp/project_include.cmake /home/lee/esp/esp-idf/components/unity/CMakeLists.txt /home/lee/esp/esp-idf/components/unity/unity/.git /home/lee/esp/esp-idf/components/vfs/CMakeLists.txt /home/lee/esp/esp-idf/components/wear_levelling/CMakeLists.txt /home/lee/esp/esp-idf/components/wifi_provisioning/CMakeLists.txt /home/lee/esp/esp-idf/components/wpa_supplicant/CMakeLists.txt /home/lee/esp/esp-idf/components/xtensa/CMakeLists.txt /home/lee/esp/esp-idf/examples/build_system/cmake/import_lib/main/lib/tinyxml2/.git /home/lee/esp/esp-idf/examples/peripherals/secure_element/atecc608_ecdsa/components/esp-cryptoauthlib/.git /home/lee/esp/esp-idf/tools/cmake/build.cmake /home/lee/esp/esp-idf/tools/cmake/component.cmake /home/lee/esp/esp-idf/tools/cmake/crosstool_version_check.cmake /home/lee/esp/esp-idf/tools/cmake/dfu.cmake /home/lee/esp/esp-idf/tools/cmake/git_submodules.cmake /home/lee/esp/esp-idf/tools/cmake/idf.cmake /home/lee/esp/esp-idf/tools/cmake/kconfig.cmake /home/lee/esp/esp-idf/tools/cmake/ldgen.cmake /home/lee/esp/esp-idf/tools/cmake/project.cmake /home/lee/esp/esp-idf/tools/cmake/project_description.json.in /home/lee/esp/esp-idf/tools/cmake/targets.cmake /home/lee/esp/esp-idf/tools/cmake/third_party/GetGitRevisionDescription.cmake /home/lee/esp/esp-idf/tools/cmake/third_party/GetGitRevisionDescription.cmake.in /home/lee/esp/esp-idf/tools/cmake/toolchain-esp32.cmake /home/lee/esp/esp-idf/tools/cmake/utilities.cmake /home/lee/esp/esp-idf/tools/cmake/version.cmake /home/lee/esp/esp-idf/tools/kconfig_new/confgen.py /home/lee/esp/esp-idf/tools/kconfig_new/config.env.in /home/lee/esp/esp-idf/tools/kconfig_new/kconfiglib.py CMakeCache.txt CMakeFiles/3.13.4/CMakeASMCompiler.cmake CMakeFiles/3.13.4/CMakeCCompiler.cmake CMakeFiles/3.13.4/CMakeCXXCompiler.cmake CMakeFiles/3.13.4/CMakeSystem.cmake CMakeFiles/git-data/grabRef.cmake app-flash_args bootloader-flash_args bootloader-prefix/tmp/bootloader-cfgcmd.txt.in config/sdkconfig.cmake config/sdkconfig.h esp-idf/bootloader/bootloader-flash_args.in esp-idf/esptool_py/app-flash_args.in esp-idf/esptool_py/flasher_args.json.in esp-idf/partition_table/partition_table-flash_args.in flash_args flash_args.in ldgen_libraries.in
./build/CMakeCache.txt://Program used to build from build.ninja files.
./build/CMakeCache.txt:CMAKE_MAKE_PROGRAM:FILEPATH=/home/lee/.espressif/tools/ninja/1.9.0/ninja
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_101d1"
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_4ba82"
./build/CMakeFiles/CMakeOutput.log:  ignore line: [Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_4ba82"]
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_c4bec"
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_983f4"
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_85b36"
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_815cd"
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_bc628"
./build/CMakeFiles/CMakeOutput.log:  ignore line: [Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_bc628"]
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_27c2e"
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_8e588"
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_5100c"
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_29c07"
./build/CMakeFiles/CMakeOutput.log:Run Build Command:"/home/lee/.espressif/tools/ninja/1.9.0/ninja" "cmTC_dffd1"
./build/rules.ninja:# It is included in the main 'build.ninja'.
./build/rules.ninja:  command = /home/lee/.espressif/tools/ninja/1.9.0/ninja -t clean
./build/rules.ninja:  command = /home/lee/.espressif/tools/ninja/1.9.0/ninja -t targets

```
Thanks 

Lee

**Comment by kolipakakondal** [comment]
It's because CMakeCache.txt has old entries, it's suggested to delete the build folder and rebuild in case of project rename/duplicate

**Comment by kolipakakondal** [comment]
Hello, All these problems were fixed with v2.4.0(latest v2.4.2). Please get the latest changes using the update site and report us back if you have any problems with this.

Thank you.
